### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by tsohleDev**
# Rspec unit test

In this PR we write unit tests for each method in:
- Person
- Classroom
- Student
- Teacher
- Rental
- Book

### CODE REVIEW

File: spec/person_specs.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      person.age = 17
+      expect(person.can_use_services?).to eq(true)
+    end
+
+    it 'returns false if the person is not of age and does not have parent permission' do
+      person.age = 17
+      person.instance_variable_set(:@parent_permission, false)
+      expect(person.can_use_services?).to eq(false)
+    end
+  end
```
Comment (Ghiftee) [inline spec/person_specs.rb:63]: - [ ] When I run these tests, I get the error below. Kindly fix it
Screenshot 2023-03-16 at 09 49 48

File: spec/classroom_specs.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      classroom.add_student(student1)
+      expect(student1.classroom).to eq(classroom)
+    end
+
+    it 'should not add the same student to the students array twice' do
+      classroom.add_student(student1)
+      expect do
+        classroom.add_student(student1)
+      end.to change { classroom.students.count }.by(0)
+    end
```
Comment (Ghiftee) [inline spec/classroom_specs.rb]: - [ ] When I run these tests, I get the error below, kindy fix it
Screenshot 2023-03-16 at 09 48 44

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @tsohleDev and @george827  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Unit tests are added
✔️ Most tests pass
✔️ Gitflow is used

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._
- [ ] Kindly update the details of your README to match this milestone especially in sections like ```usuage``` and ```future features``` as this would make your documentation more professional

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by brytebee — APPROVED** [review]
Hi, @tsohleDev & @george827👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great use or rspec.✔️
- Test pass and are green.✔️
- Great implementing the suggested changes.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**